zio-http/src/test/scala/zio/http/TestServer.scala

```diff
@@ -0,0 +1,155 @@
+package zio.http
+
+import zio._
+import zio.http.Server.ErrorCallback
+
+/**
+ * Enables tests that make calls against "localhost" with user-specified
+ * Behavior/Responses.
+ *
+ * @param state
+ *   State that can be consulted by our behavior PartialFunction
+ * @param behavior
+ *   Describes how the Server should behave during your test
+ * @param driver
+ *   The web driver that accepts our Server behavior
+ * @param bindPort
+ *   Port for HTTP interactions
+ * @tparam State
+ *   The type of state that will be mutated
+ */
+final case class TestServer[State](
+  state: Ref[State],
+  behavior: PartialFunction[(State, Request), (State, Response)],
+  driver: Driver,
+  bindPort: Int,
+) extends Server {
+
+  /**
+   * Define 1-1 mappings between incoming Requests and outgoing Responses
+   *
+   * @param expectedRequest
+   *   Request that will trigger the provided Response
+   * @param response
+   *   Response that the Server will send
+   *
+   * @example
+   *   {{{ ZIO.serviceWithZIO[TestServer[Unit]]( _.addRequestResponse(
+   *   Request.get(url = URL.root.setPort(port = ???)), Response(Status.Ok)
+   *   ).install ) }}}
+   *
+   * @return
+   *   The TestSever with new behavior.
+   */
+  def addRequestResponse(
```

Comment:
It might be useful to allow the user to perform `assertTrue` on the request, and to generate the response from the request.

Also, I can see these functions return new `TestServer`. This is not consistent with `TestClock`, `TestSystem`, `TestRandom`, etc. 

See, for example, `TestConsole.feedLines` and friends, which modify internal state rather than return a new test construct. I think all of these methods should **not** return a new `TestServer`, but rather, should modify internal state, to be consistent with other test control structures.

Reply:
Rodger. An earlier version of this used this approach, but then we ended up changing to this style at the last minute. I will rework it to be more in line with those other services.